net-p2p/biglybt/biglybt-3.8.0.0.ebuild

```diff
@@ -0,0 +1,112 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# javadocs are too broken and no tests, #839681
+JAVA_PKG_IUSE="source"
+MAVEN_PROVIDES="com.${PN}:${PN}-core:${PV} com.${PN}:${PN}-ui:${PV}"
+
+inherit desktop java-pkg-2 java-pkg-simple optfeature xdg
+
+DESCRIPTION="Feature-filled Bittorrent client based on the Azureus open source project"
+HOMEPAGE="https://www.biglybt.com"
+SRC_URI="https://github.com/BiglySoftware/BiglyBT/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/BiglyBT-${PV}"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc64"
+
+CP_DEPEND="
+	dev-java/commons-cli:0
+	dev-java/swt:4.27[cairo]
```

Comment:
Could it use even the latest [swt](https://github.com/gentoo/gentoo/commit/bd15678434a8597a0f9d265c8c04abd67c120b84) or is it llimited to certain versions of `*.so` files?

Reply:
> From my limited testing it works with our latest. Also their [pom](https://github.com/BiglySoftware/BiglyBT/blob/29fb3b1d0eebb9611dcc26c16aa521d926246031/pom.xml#L15) leads me to believe it will work with anything above 4.9. Should I swap the dep to `>=dev-java/swt-4.9[cairo]`?

Sorry for my late response. Putting dependency without slot is not an option for java packages. 
If `biglybt` really works with any (of those many slottet) version of `dev-java/swt`, without any runtime errors, we might add [dev-java/swt-4.35](https://github.com/gentoo/gentoo/commit/337562182540cfe758558cca129afa0ac6477674) to `SLOT=0` and have `biglybt` CP_DEPEND on `>=dev-java/swt-4.35:0[cairo]`. Then cleanout all the older unused slots.

Reply:
> > Putting dependency without slot is not an option for java packages.
> 
> I see.
> 
> Doesn't also dev-java/openjfx/ depend on swt? Before such change I think it also should be tested.

Sure, it does. [Both versions](https://packages.gentoo.org/packages/dev-java/openjfx) depend on slot `4.10``dev-java/swt:4.10`. And for switching it to a newer `swt`, there was [another discussion](https://github.com/gentoo/gentoo/pull/36794#issuecomment-2626668054). 
It turned out openjfx would also need to get switched from `gtk+2` -> `gtk+3` which is completely out of my scope (no clue about that stuff).